jgrapht-core/src/main/java/org/jgrapht/alg/matching/Edge.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching;
+
+import org.jgrapht.util.FibonacciHeapNode;
+
+/**
+ * This class is a supporting data structure for Kolmogorov's Blossom V algorithm. Represents an edge
+ * between two nodes. Each edge has direction. According to this direction it is present in two circular
+ * doubly linked lists of incident edges. The references to the next and previous edges of this list
+ * are maintained via {@link Edge#next} and {@link Edge#prev}
+ *
+ * @author Timofey Chudakov
+ * @see KolmogorovMinimumWeightPerfectMatching
+ * @since June 2018
+ */
+class Edge {
```

Comment:
Avoid using such generic names as Edge, Node, Tree, State, and Initializer since this package is not dedicated to just your algorithm.  So either use more specific names, or namespace them (e.g. in a dedicated subpackage).